evalml/tests/component_tests/test_time_series_imputer.py

```diff
@@ -159,7 +169,7 @@ def test_categorical_and_numeric_input(imputer_test_data):
     y = pd.Series([0, 0, 1, 0, 1])
     imputer = TimeSeriesImputer()
     imputer.fit(X, y)
-    transformed, _ = imputer.transform(X, y)
+    transformed, _ = imputer.transform(X.ww.copy(), y)
```

Comment:
This was needed because `_handle_nullable_types` actually changes the input data, so the second imputer calls were being run without any nullable types. 

I want to triple check that `_handle_nullable_types` mutating the input data isn't problematic. I seem to remember having a conversation in which I learned that evalml as a whole doesn't worry too much about not mutating input data, but I'm still worried this will prove to be problematic at the pipeline level somehow.

Reply:
The fact that `_handle_nullable_types` mutates the input data was hiding some incompatibilities in predict/transform methods, because my `test_lgbm_with_nullable_types` test was passing in an `X` without nullable types to predict and predict_proba 🙃. 

I still can't think of a reason good enough to copy the data every time. I think we just need to be aware of this in tests that are checking that nullable types can be passed into components, and in transformers, make sure we're maintaining the original types in the transformed data.

Reply:
This is a great point and edge case - could you add some comments to the tests where you call copy for posterity?

Reply:
added comments: 827305987